packages/components/src/box-control/index.js

```diff
@@ -57,7 +57,7 @@ export default function BoxControl( {
 	resetValues = DEFAULT_VALUES,
 	allowNegativeValues = false,
 } ) {
-	inputProps = allowNegativeValues ? { min: -Infinity } : defaultInputProps;
+	inputProps.min = allowNegativeValues ? -Infinity : defaultInputProps.min;
```

Comment:
Thanks for following up, Justin!

I'm actually wondering about this new prop. If a dev does something like this:
```jsx
<BoxControl
    inputProps={ { min: -10 } }
    allowNegativeValues
/>
```
At first glance this looks correct, but the minimum of `-10` is actually being overwritten still with `-Infinity` by this code.

There are probably ways to fix that, but then another scenario is something meaningless like:
```jsx
<BoxControl
    inputProps={ { min: 5 } }
    allowNegativeValues
/>
```

I think it might be best not to have two props that can do the same thing. Maybe there's no need for the `allowNegativeValues` prop at all, and the `BoxControl` used for margin could be simplified to:
```jsx
<BoxControl
    inputProps={ { min: -Infinity } }
/>
```

It isn't as clean as `allowNegativeValues`, but I think it's less likely to cause confusion.